psd.h

```diff
@@ -201,6 +231,180 @@ namespace psd {
     ChannelID channelID;
     uint64_t length;
   };
+  
+  struct OSType {
+    virtual OSTypeKey type() const = 0;
+    ~OSType(){}
+  };
+
+  struct OSTypeClassMetaType {
+    uint32_t keyClassID = 0;
+    std::string name;
+  };
+
+  struct OSTypeUnitFloat: OSType {
+    enum class Unit {
+      Angle = PSD_DEFINE_DWORD('#', 'A', 'n', 'g'), // base degrees
+      Density = PSD_DEFINE_DWORD('#', 'R', 's', 'l'), // base per inch
+      Distance = PSD_DEFINE_DWORD('#', 'R', 'l', 't'), // base 72ppi
+      None = PSD_DEFINE_DWORD('#', 'N', 'n', 'e'), // coerced
+      Percent = PSD_DEFINE_DWORD('#', 'P', 'r', 'c'), // unit value
+      Pixel = PSD_DEFINE_DWORD('#', 'P', 'x', 'l'), // tagged unit value
+    };
+
+    Unit   unit;
+    double value = 0.0;
+
+    OSTypeUnitFloat(Unit u, double v): unit{u}, value{v}{}
+    OSTypeKey type() const override {
+      return OSTypeKey::UnitFloat;
+    }
+  };
+
+  struct OSTypeDouble: OSType {
+    double value = 0.0;
+
+    OSTypeDouble(double v): value{v}{}
+    OSTypeKey type() const override {
+      return OSTypeKey::Double;
+    }
+  };
+
+  struct OSTypeClass: OSType {
+    std::wstring        className;
+    OSTypeClassMetaType meta;
+
+    OSTypeKey type() const override {
+      return OSTypeKey::ClassType;
+    }
+  };
+
+  struct OSTypeString: OSType {
+    std::wstring value;
+
+    OSTypeString(std::wstring const &v): value{v}{}
```

Comment:
Use this kind of style in all places:
```suggestion
    OSTypeString(const std::wstring& v): value(v) { }
```